### PR TITLE
Choose evaluation example from test set

### LINKED ISSUE #796: Harcoded incorrect (and repeated) validation example
👋 I have found this code to be surprisingly incorrect as my dataset has nothing to do with this validation example. Its also repeated in every single finetuning script.

```python
# produce an example:
    instruction = "Recommend a movie for me to watch during the weekend and explain the reason."
    fabric.print(instruction)
    sample = {"instruction": instruction, "input": ""}
    prompt = generate_prompt(sample)
    encoded = tokenizer.encode(prompt, device=fabric.device)
    with fabric.init_tensor():
        # do not set `max_seq_length=max_returned_token` because memory is not a concern here
        model.set_kv_cache(batch_size=1)
    output = generate(model, encoded, max_returned_tokens=len(encoded) + eval_max_new_tokens, temperature=0.8)
    model.clear_kv_cache()
    output = tokenizer.decode(output)
    fabric.print(output)

```

### LINKED ISSUE #1443: validation output during finetuning
I am doing full finetuning using my custom dataset, every time the validation outputs something like this:
image
My dataset isn't about movies, is this a bug?

### PR DESCRIPTION
**Pull request by rasbt**
Instead of using the "Recommend what movie to watch" instruction in all finetuning scripts, we now select an example from the test set for the evaluation print during finetuning.

The default can be overridden via the following setting:

```bash
litgpt finetune ... --eval.evaluate_example "random" | "first" | 99
```

Fixes #796
Fixes #1443

### CODE REVIEW

File: litgpt/args.py [hunk 1]
```diff
@@ -85,3 +85,8 @@ class EvalArgs:
     """Whether to evaluate on the validation set at the beginning of the training"""
     final_validation: bool = True
     """Whether to evaluate on the validation set at the end of the training"""
+    final_validation: bool = True
+    """Whether to evaluate on the validation set at the end of the training"""
```
Comment (rasbt) [inline litgpt/args.py:89]:
```suggestion
```
